File: app/src/main/java/com/mnj/java_solutions/gfg/LeadersInArray.java
package com.mnj.java_solutions.gfg;

import java.util.Stack;

public class LeadersInArray {

    public void leadersInArray() {

        int[] a = {16, 17, 4, 3, 5, 2};

        Stack<Integer> stack = new Stack<>();
        stack.push(a[a.length-1]);

        for (int i = a.length-1; i >= 0; i--) {

            if (stack.peek() < a[i]) {
                stack.push(a[i]);
            }
        }
        while (!stack.isEmpty()) {
            System.out.println("==>> Leaders are:" + stack.pop());
        }
    }
}

File: app/src/main/java/com/mnj/java_solutions/gfg/LeadersInArray.kt
package com.mnj.java_solutions.gfg

import java.util.*

class LeadersInArray {
    fun leadersInArray() {
        val a = intArrayOf(16, 17, 4, 3, 5, 2)
        val stack = Stack<Int>()
        stack.push(a[a.size - 1])
        for (i in a.indices.reversed()) {
            if (stack.peek() < a[i]) {
                stack.push(a[i])
            }
        }
        while (!stack.isEmpty()) {
            println("==>> Leaders are:" + stack.pop())
        }
    }
}